core/src/com/unciv/logic/HexMath.kt

```diff
@@ -194,4 +196,34 @@ object HexMath {
 
     fun getClockDirectionToWorldVector(clockDirection: Int): Vector2 =
         clockToWorldVectors[clockDirection] ?: Vector2.Zero
+
+    fun getDistanceFromEdge(vector: Vector2, mapParameters: MapParameters): Int {
+        val x = vector.x.toInt()
+        val y = vector.y.toInt()
+        if (mapParameters.shape == MapShape.rectangular) {
+            val height = mapParameters.mapSize.height
+            val width = mapParameters.mapSize.width
+            val left = if (mapParameters.worldWrap) Int.MAX_VALUE else width / 2 - (x - y)
+            val right = if (mapParameters.worldWrap) Int.MAX_VALUE else (width - 1) / 2 - (y - x)
+            val top = height / 2 -  (x + y) / 2
+            // kotlin's Int division rounds in different directions depending on sign! Thus 1 extra `-1`
+            val bottom = (x + y - 1) / 2 + (height - 1) / 2
+            return min(min(left, right), min(top, bottom))
```

Comment:
Can't min take parameters as needed? Do we really need nesting?

Reply:
Not that I know... There's surely some kotlin extension emulating SQL for such - but - it's just the one place and I doubt the bytecode could get better. At least I didn't nest simply left to right! A generic fun for that might be fun to do... Can you _slice_ varargs?

Reply:
Varargs is in the end an array, and you can inputvarargs with the .. operator, so yes
I think in tileSetStrings I played around with this a little

Reply:
... and the kotlin lib already has it. called `minOf`. next time.